synapse/visibility.py

```diff
@@ -587,7 +599,7 @@ async def filter_events_for_server(
         The filtered events.
     """
 
-    def is_sender_erased(event: EventBase, erased_senders: Dict[str, bool]) -> bool:
+    def is_sender_erased(event: EventBase, erased_senders: Mapping[str, bool]) -> bool:
```

Comment:
Prefer to take immutable types as arguments where possible---it makes the reader's life much easier!